### ISSUE TITLE
Rustfmt mangles macro input, replacing ':' with '='

### ISSUE THREAD

**Issue by daboross** [rust-lang/rustfmt#3660]
I use this macro in one of my projects, based off of one of serde's macros, which takes a struct and a number of possible trait bounds to use for that struct. When formatted, the macro calls are mangled. My original code:

```rust
seq_impl!(Option<T: TransferOwnership>);
seq_impl!(Vec<T: TransferOwnership>);
seq_impl!(VecDeque<T: TransferOwnership>);
seq_impl!(LinkedList<T: TransferOwnership>);
seq_impl!(HashSet<T: TransferOwnership + Hash + Eq, S: BuildHasher>);
seq_impl!(BTreeSet<T: Ord + TransferOwnership>);
seq_impl!(BinaryHeap<T: Ord + TransferOwnership>);
```

The formatted code:

```rust
seq_impl!(Option<T = TransferOwnership>);
seq_impl!(Vec<T = TransferOwnership>);
seq_impl!(VecDeque<T = TransferOwnership>);
seq_impl!(LinkedList<T = TransferOwnership>);
seq_impl!(HashSet<T: TransferOwnership + Hash + Eq, S: BuildHasher>);
seq_impl!(BTreeSet<T = Ord + TransferOwnership>);
seq_impl!(BinaryHeap<T = Ord + TransferOwnership>);
```

The colons are replaced with `=`, which is invalid for this macro invocation.

This occurs even without any definition of the macro, so I've not included it. My test case is simply the following file:

```rust
fn main() {
    seq_impl!(Option<T: TransferOwnership>);
    seq_impl!(Vec<T: TransferOwnership>);
    seq_impl!(VecDeque<T: TransferOwnership>);
    seq_impl!(LinkedList<T: TransferOwnership>);
    seq_impl!(HashSet<T: TransferOwnership + Hash + Eq, S: BuildHasher>);
    seq_impl!(BTreeSet<T: Ord + TransferOwnership>);
    seq_impl!(BinaryHeap<T: Ord + TransferOwnership>);
}
```

This occurs running `rustfmt 1.3.0-nightly (d334502 2019-06-09)`, with the `--edition 2018` flag, and no other configuration.

**Comment by calebcartwright** [comment]
I think this _may_ also be fixed in #3661

**Comment by daboross** [comment]
#3661 does fix this - thanks!

I'm surprised that this macro input is somehow interpreted as an associated type bound, but I guess that makes sense. It's just a regular type bound, but I guess that would be pretty hard to tell without looking into the macro source.
